src/analyzer.ts

```diff
@@ -481,17 +481,99 @@ export class Analyzer {
     const uniqueDependencies = new Map<string, JsonAstNode>();
     const children = dependenciesAst.children ?? [];
     for (let i = 0; i < children.length; i++) {
+      // A dependency can be either a plain string, or an object with a "script"
+      // property plus optional extra annotations.
       const maybeUnresolved = children[i];
-      const stringResult = failUnlessNonBlankString(
-        maybeUnresolved,
-        packageJson.jsonFile
-      );
-      if (!stringResult.ok) {
+      let specifierResult;
+      let soft = false; // Default;
+      if (maybeUnresolved.type === 'string') {
+        specifierResult = failUnlessNonBlankString(
+          maybeUnresolved,
+          packageJson.jsonFile
+        );
+        if (!specifierResult.ok) {
+          encounteredError = true;
+          placeholder.failures.push(specifierResult.error);
+          continue;
+        }
+      } else if (maybeUnresolved.type === 'object') {
+        specifierResult = findNodeAtLocation(maybeUnresolved, ['script']);
+        if (specifierResult == null) {
```

Comment:
Assuming this is specifically `==` to catch both `undefined` and `null`?

Reply:
I copied this style from the other code @rictic wrote in this file. I have been in the habit of using `=== undefined` when we know we don't need to handle `null` (a habit I learned from @justinfagnani) -- but maybe @rictic has a reason to use `== null` here?

Reply:
I use `== null` to signify that either we do need to handle both (good), or that I'm not sure (bad - and I leave a TODO).

Reply:
Switched everything to check `=== undefined` when the type indicated that it could not be null (which is almost everything).